sql/src/test/java/io/crate/integrationtests/TableStatsServiceIntegrationTest.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.integrationtests;
+
+import io.crate.metadata.TableIdent;
+import io.crate.planner.TableStats;
+import io.crate.testing.UseRandomizedSchema;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.is;
+
+
+@ESIntegTestCase.ClusterScope(supportsDedicatedMasters = false, numDataNodes = 2, numClientNodes = 0)
+public class TableStatsServiceIntegrationTest extends SQLTransportIntegrationTest {
+
+    @Before
+    public void setRefreshInterval() {
+        execute("set global transient stats.service.interval='10ms'");
+    }
+
+    @After
+    public void setResetInterval() {
+        execute("reset global stats.service.interval");
+    }
+
+    @Test
+    public void testStatsUpdated() throws Exception {
+        execute("create table t1(a int) with (number_of_replicas = 1)");
+        ensureGreen();
+        execute("insert into t1(a) values(1), (2), (3), (4), (5)");
+        execute("refresh table t1");
+        assertBusy(() -> {
+                TableStats tableStats = internalCluster().getDataNodeInstance(TableStats.class);
+                assertThat(tableStats.numDocs(new TableIdent(sqlExecutor.getDefaultSchema(), "t1")), is(5L));
```

Comment:
we should also check sizeInBytes (at least > 0)

Reply:
I tried to but it gets >0 after several seconds... any idea why?

Reply:
hm yes, ShardSizeExpression memoize the size for 10sec...  so nothing we can do here besides a sleep or making this memoize period configurable :-( ok lets leave it out